src/components/tag/tag.js

```diff
@@ -202,97 +140,96 @@ TagNormalBody.propTypes = {
   }).isRequired,
 };
 
-const Tag = props => (
-  <Constraints.Horizontal constraint={props.horizontalConstraint}>
-    <div
-      css={theme => [
-        css`
-          min-width: 0;
-          display: flex;
-          background-color: ${getWrapperBackgroundColor(props.type, theme)};
-        `,
-      ]}
-    >
-      {props.linkTo ? (
-        <TagLinkBody
-          styles={props.styles}
-          type={props.type}
-          onClick={props.onClick}
-          onRemove={props.onRemove}
-          linkTo={props.linkTo}
-          isDisabled={props.isDisabled}
-        >
-          {props.children}
-        </TagLinkBody>
-      ) : (
-        <TagNormalBody
+const Tag = props => {
+  const linkProps =
+    props.linkTo && !props.isDisabled ? { as: Link, to: props.linkTo } : {};
+
+  return (
+    <Constraints.Horizontal constraint={props.horizontalConstraint}>
+      <div
+        css={theme =>
+          css`
+            a {
+              cursor: pointer;
+              text-decoration: none;
+            }
+            cursor: default;
+            min-width: 0;
+            display: flex;
+            background-color: ${getWrapperBackgroundColor(props.type, theme)};
+          `
+        }
+      >
+        <TagBody
+          {...linkProps}
           styles={props.styles}
           type={props.type}
           onClick={props.onClick}
           onRemove={props.onRemove}
           isDisabled={props.isDisabled}
         >
           {props.children}
-        </TagNormalBody>
-      )}
-      {Boolean(props.onRemove) && (
-        <AccessibleButton
-          label="Remove"
-          isDisabled={props.isDisabled}
-          onClick={props.isDisabled ? undefined : props.onRemove}
-          css={theme => {
-            const overwrittenVars = {
-              ...vars,
-              ...theme,
-            };
+        </TagBody>
 
-            return [
-              css`
-                border-color: ${props.type === 'warning'
-                  ? overwrittenVars[designTokens.borderColorForTagWarning]
-                  : overwrittenVars[designTokens.borderColorForTag]};
-                padding: 0 ${vars.spacingXs};
-                border-radius: 0
-                  ${overwrittenVars[designTokens.borderRadiusForTag]}
-                  ${overwrittenVars[designTokens.borderRadiusForTag]} 0;
-                display: flex;
-                align-items: center;
-                background: inherit;
-                border-style: solid;
-                border-width: 1px 1px 1px 1px;
-                &:hover,
-                &:focus {
-                  border-color: ${overwrittenVars[
-                    designTokens.borderColorForTagWarning
-                  ]};
+        {Boolean(props.onRemove) && (
+          <AccessibleButton
+            label="Remove"
+            isDisabled={props.isDisabled}
+            onClick={props.isDisabled ? undefined : props.onRemove}
```

Comment:
We no longer need to do this, since now the `AccessibleButton` takes care of "ignoring" onClicks when disabled, right?